src/cmds/fs/sort.c

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file
+ * @brief Sort lines of text files to standart output.
+ *
+ * @date 28.09.18
+ * @author Mark Kovalev
+ */
+
+#include <unistd.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+
+
+static int blank, ign_case, rev;
+
+static void print_usage(void) {
+	printf("Usage: sort [OPTION]... [FILE]...\n"
+		"Write sorted lines of FILE to standard output.\n"
+		"\nOrdering options:\n\n"
+		"  -b\tignore leading blanks\n"
+		"  -f\tfold lower case to upper case characters\n"
+		"  -r\treverse the result of comparisons\n"
+		"\nOther options:\n\n"
+		"  -h\tdisplay this help and exit\n"
+		"\n*** WARNING ***\n"
+		"The locale specified by the environment affects sort order.\n"
+		"This implementation assumes that LC_ALL=C set to get the\n"
+		"traditional sort order that uses native byte values.\n");
+}
+
+static int compare(char *fst, char *snd) {
+    int i = 0, j = 0, bigger = -1;
+    char f, s;
+
+    if (blank) {
+        while (fst[i] == ' ')
+            ++i;
+        while (snd[j] == ' ')
+            ++j;
+    }
+
+    while (fst[i] != '\0' && snd[j] != '\0') {
+        if (ign_case) {
+            if (((64 < fst[i] && fst[i] < 91) || (96 < fst[i] && fst[i] < 123)) &&
+                ((64 < snd[j] && snd[j] < 91) || (96 < snd[j] && snd[j] < 123)) &&
+                (fst[i] - snd[j] == 32 || fst[i] - snd[j] == -32))
+                bigger = (bigger == -1) ? (fst[i] > snd[j]) : bigger;
+
+            f = (96 < fst[i] && fst[i] < 123) ? fst[i] - 32 : fst[i];
+            s = (96 < snd[j] && snd[j] < 123) ? snd[j] - 32 : snd[j];
+            if (f != s)
+                return f > s;
+        }
+        else if (fst[i] != snd[j])
+            return (fst[i] > snd[j]);
+        ++i;
+        ++j;
+    }
+
+    if (bigger != -1 && fst[i] == snd[j])
+        return bigger;
+
+    return (fst[i] > snd[j]);
+}
+
+static void merge(char **in_strs, int begin, int mid, int end, char **buff_strs) {
+    int i = begin, k = begin, l = mid + 1;
+
+    while (k <= mid && l <= end) {
+        buff_strs[i++] = (compare(in_strs[k], in_strs[l])) ? in_strs[l++] : in_strs[k++];
+    }
+
+    while (k <= mid)
+        buff_strs[i++] = in_strs[k++];
+    while (l <= end)
+        buff_strs[i++] = in_strs[l++];
+
+    for(i = begin; i <= end; ++i)
+        in_strs[i] = buff_strs[i];
+}
+
+static void sort(char **in_strs, int begin, int end, char **buff_strs) {
```

Comment:
@oniich First of all, thank you for the contribution!
Now, as far as I can see you implemented Merge Sort inside the command. But it is better to use some existing stuff if it is presented. In your case, you could try libc/stdlib/qsort.c. It's from standard C library. If it will work, it would be a better choice.
In that case all you need is to use qsort, and write your own string comparator for it.